File: ios/RNGiphySDK.m
#import <React/RCTBridgeModule.h>
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_REMAP_MODULE(GiphyReactNativeSDK, RNGiphySDK, RCTEventEmitter)

RCT_EXTERN_METHOD(configure: (NSDictionary)config)

@end

File: ios/RNGiphySDK.swift
import GiphyUISDK

@objc(RNGiphySDK)
class RNGiphySDK: NSObject {
  @objc
  public static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc(configure:)
  func configure(config: NSDictionary) -> Void {
    if let apiKey = config["apiKey"] as? String {
      let verificationMode = config["verificationMode"] as? Bool ?? false
      let metadata = [RNSDKInfo.shared.name: RNSDKInfo.shared.version ?? ""]
      Giphy.configure(
        apiKey: apiKey,
        verificationMode: verificationMode,
        metadata: metadata
      )
    }
  }
}
